src/libawkward/io/json.cpp

```diff
@@ -200,16 +222,27 @@ namespace awkward {
     rj::PrettyWriter<rj::StringBuffer> writer_;
   };
 
-  ToJsonPrettyString::ToJsonPrettyString(int64_t maxdecimals)
-      : impl_(new ToJsonPrettyString::Impl(maxdecimals)) { }
+  ToJsonPrettyString::ToJsonPrettyString(int64_t maxdecimals,
+    const char* nan_string,
+    const char* infinity_string,
+    const char* minus_infinity_string)
+      : impl_(new ToJsonPrettyString::Impl(maxdecimals))
+      , nan_string_(nan_string)
+      , infinity_string_(infinity_string)
+      , minus_infinity_string_(minus_infinity_string) { }
 
   ToJsonPrettyString::~ToJsonPrettyString() {
     delete impl_;
   }
 
   void
   ToJsonPrettyString::null() {
-    impl_->null();
+    if (nan_string_ != nullptr) {
+      impl_->string(nan_string_, (int64_t)strlen(nan_string_));
+    }
+    else {
+      impl_->null();
+    }
```

Comment:
For the same reason as above, please revert this.

Reply:
done